Userland/Libraries/LibJS/Runtime/Intrinsics.cpp

```diff
@@ -331,6 +325,24 @@ JS_ENUMERATE_TEMPORAL_OBJECTS
 
 #undef __JS_ENUMERATE_INNER
 
+#define __JS_ENUMERATE(ClassName, snake_name)                                   \
+    void Intrinsics::initialize_##snake_name##_object()                         \
+    {                                                                           \
+        VERIFY(!m_##snake_name##_object);                                       \
+        m_##snake_name##_object = heap().allocate<ClassName>(m_realm, m_realm); \
+    }                                                                           \
+                                                                                \
+    ClassName* Intrinsics::snake_name##_object()                                \
+    {                                                                           \
+        if (!m_##snake_name##_object)                                           \
+            initialize_##snake_name##_object();                                 \
+        return m_##snake_name##_object;                                         \
+    }
```

Comment:
NBD, but these could just be one function. It's separated above because constructors and prototypes need to do the same work.

```c++
    ClassName* Intrinsics::snake_name##_object()                                    \
    {                                                                               \
        if (!m_##snake_name##_object)                                               \
            m_##snake_name##_object = heap().allocate<ClassName>(m_realm, m_realm); \
        return m_##snake_name##_object;                                             \
    }
```

Reply:
Yeah, I wasn't sure whether to do it like this or keep it consistent with a separate `initialize` method. Done :+1: